controllers/config.go

```diff
@@ -0,0 +1,57 @@
+package controllers
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"path/filepath"
+
+	configv1 "github.com/openshift/api/config/v1"
+)
+
+// imagesReference allows build systems to inject imagesReference for CCCMO components
+type imagesReference struct {
+	CloudControllerManagerAWS       string `json:"cloudControllerManagerAWS"`
+	CloudControllerManagerOpenStack string `json:"cloudControllerManagerOpenStack"`
+}
+
+// operatorConfig contains configuration values for templating resources
+type operatorConfig struct {
+	ManagedNamespace string
+	ControllerImage  string
+}
+
+func getProviderFromInfrastructure(infra *configv1.Infrastructure) (configv1.PlatformType, error) {
+	if infra == nil || infra.Status == (configv1.InfrastructureStatus{}) || infra.Status.PlatformStatus == nil {
+		return "", fmt.Errorf("platform status is not pupulated on infrastructure")
+	}
```

Comment:
Nit: We don't need the middle part of this as it's not a pointer, deref will be fine
```suggestion
	if infra == nil || infra.Status.PlatformStatus == nil {
		return "", fmt.Errorf("platform status is not pupulated on infrastructure")
	}
```